cobra/flux_analysis/parsimonious.py

```diff
@@ -11,53 +17,90 @@ def optimize_minimal_flux(model, already_irreversible=False,
     form it was in before. See http://dx.doi.org/10.1038/msb.2010.47
 
-    model : :class:`~cobra.core.Model` object
+    cobra_model : :class:`~cobra.core.Model` object
 
     already_irreversible : bool, optional
         By default, the model is converted to an irreversible one.
         However, if the model is already irreversible, this step can be
         skipped.
 
+    fraction_of_optimum : float, optional
+        Fraction of optimum which must be maintained. The original objective
+        reaction is constrained to be greater than maximal_value *
+        fraction_of_optimum. By default, this option is specified to be 1.0
+
+    desired_objective_value : float, optional
+        A desired objective value for the minimal solution that bypasses the
+        initial optimization result.
+
+    solver : string of solver name
+        If None is given, the default solver will be used.
+
+    Updates everything in-place, returns model to original state at end.
     """
-    if not already_irreversible:
-        modify.convert_to_irreversible(model)
-    model.optimize(**optimize_kwargs)
-    # if the problem is infeasible
-    if model.solution.f is None:
-        raise Exception("model could not be solved")
-    old_f = model.solution.f
-    old_objective_coefficients = {}
-    old_lower_bounds = {}
-    old_upper_bounds = {}
-    for reaction in model.reactions:
-        # if the reaction has a nonzero objective coefficient, then
-        # the same flux should be maintained through that reaction
-        if reaction.objective_coefficient != 0:
-            old_objective_coefficients[reaction] = \
-                reaction.objective_coefficient
-            old_lower_bounds[reaction] = reaction.lower_bound
-            old_upper_bounds[reaction] = reaction.upper_bound
-            x = model.solution.x_dict[reaction.id]
-            reaction.lower_bound = x
-            reaction.upper_bound = x
-            reaction.objective_coefficient = 0
-        else:
-            reaction.objective_coefficient = 1
-    # set to minimize flux
-    optimize_kwargs["objective_sense"] = "minimize"
-    model.optimize(**optimize_kwargs)
-    # make the model back the way it was
-    for reaction in model.reactions:
-        if reaction in old_objective_coefficients:
-            reaction.objective_coefficient = \
-                old_objective_coefficients[reaction]
-            reaction.lower_bound = old_lower_bounds[reaction]
-            reaction.upper_bound = old_upper_bounds[reaction]
+
+    # Store objective for later use
+    original_objective = dict(cobra_model.objective)
+    if desired_objective_value:
+        assert len(original_objective) == 1
+
+    # Convert to irreversible, so all reactions will have a positive flux
+    convert_to_irreversible(cobra_model)
+
+    solver = solver_dict[get_solver_name() if solver is None else solver]
+
+    solution = None
+    try:

```

Comment:
It's better to handle this explicitly by using ```if solution.status == "optimal"```

Generally, only code which is expected to error should be put in the try/except block. Otherwise it obscures errors.